black_it/samplers/gaussian_process.py

```diff
@@ -111,49 +111,34 @@ def fit(self, X: NDArray[np.float64], y: NDArray[np.float64]) -> None:
                 RuntimeWarning,
             )
 
-        # initialize GP class from GPy with a Matern kernel by default
-        dims = X.shape[1]
-        kern = GPy.kern.Matern52(dims, variance=1.0, ARD=False)
+        # initialize GP class from scikit-learn with a Matern kernel
+        kernel = kernels.Matern(length_scale=1, length_scale_bounds=(1e-5, 1e5), nu=2.5)
+
         noise_var = y.var() * 0.01
 
-        self._gpmodel = GPRegression(
-            X, y, kernel=kern, noise_var=noise_var, mean_function=None
+        self._gpmodel = GaussianProcessRegressor(
+            kernel=kernel,
+            alpha=noise_var,
+            n_restarts_optimizer=self.optimize_restarts,
+            optimizer="fmin_l_bfgs_b",
+            random_state=self._get_random_seed(),
         )
 
-        # Make sure we do not get ridiculously small residual noise variance
-        self._gpmodel.Gaussian_noise.constrain_bounded(
-            1e-9, 1e6, warning=False
-        )  # constrain_positive(warning=False)
-
-        # we need to set the seed globally for GPy optimisations
-        # to give reproducible results
-        np.random.seed(self._get_random_seed())
-        random.seed(self._get_random_seed())
-        if self.max_iters > 0:
-            # --- update the model maximizing the marginal likelihood.
-            if self.optimize_restarts == 1:
-                self._gpmodel.optimize(
-                    optimizer="bfgs",
-                    max_iters=self.max_iters,
-                    messages=False,
-                    ipython_notebook=False,
-                )
-            else:
-                self._gpmodel.optimize_restarts(
-                    num_restarts=self.optimize_restarts,
-                    optimizer="bfgs",
-                    max_iters=self.max_iters,
-                    verbose=False,
-                )
+        self._gpmodel.fit(X, y)
+
+        # store minimum
+        if self.acquisition == "expected_improvement":
+            m, _ = self._predict_mean_std(X)
+            self._fmin = np.min(m)
 
     def predict(self, X: NDArray[np.float64]) -> NDArray[np.float64]:
         """Predict using a gaussian process surrogate model."""
         # predict mean or expected improvement on the full sample set
         if self.acquisition == _AcquisitionTypes.EI.value:
             # minus sign needed for subsequent sorting
-            candidates_score = -self._predict_EI(X)[:, 0]
+            candidates_score = -self._predict_EI(X, self.jiter)
```

Comment:
```suggestion
            candidates_score = -self._predict_EI(X, self.jitter)
```

Reply:
addressed in daac2d8fb4ec (to be squashed)